cli/cmd/profile.go

```diff
@@ -99,13 +99,13 @@ func newCmdProfile() *cobra.Command {
 			}
 
 			if options.template {
```

Comment:
Should we be reading the cluster domain from the config object?

Reply:
Yes and no:

I think there's two different use cases here:

1. I want to create my default service profile with the cluster domain read from the config (default)
2. I want to create a custom service profile with the cluster domain overwritten from cmd args

Allowing both uses cases at the same time shouldn't be a problem. I'd implement that behaviour in the (possible) next PR which introduces the cli flag to actually set a custom cluster domain.

Reply:
Why would you want to create a custom service profile with a different domain from the control plane?

Reply:
When you have multiple clusters, each with a different domain you don't want to switch into different kubectl contexts to create the appropriate service profile.

Instead it'd be easier to pass this as a cli flag so you can generate your service profile and then add it to source control.

So mostly out of convenience.

Reply:
@arminbuerkle I see what you're saying, but I think it will ultimately be confusing if the Linkerd CLI's behavior doesn't match with the current kubectl context.  I'd prefer to maintain the assumption that the Linkerd CLI should always work with relation to the current kubectl context.

Reply:
@adleong I understand, although i wouldn't agree. I think having both options can only be beneficial especially when the default option (when `--cluster-domain` cli arg is _NOT_ set) generates a service profile from the cluster config.

---

That being said: I implemented your suggestion in  8420bb3f51e0085e8aee2c3d1d3f0b6a40aac16f 

Overwriting it from CLI wouldn't be part of this PR anyways.